common/libraries/sql-lib/src/main/java/org/eclipse/dataspaceconnector/sql/SqlQueryExecutor.java

```diff
@@ -0,0 +1,321 @@
+/*
+ *  Copyright (c) 2021 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.sql;
+
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Collections;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+public final class SqlQueryExecutor {
+
+    private SqlQueryExecutor() {
+    }
+
+    /**
+     * Intended for mutating queries.
+     *
+     * @param sql       the parametrized sql query
+     * @param arguments the parameteres to interpolate with the parametrized sql query
+     * @return rowsChanged
+     * @throws SQLException if execution of the query was failing
+     */
+    public static int execute(Connection connection, String sql, Object... arguments) throws SQLException {
```

Comment:
The `SqlQueryExecutor` shall neither open nor close any connections, that's rather a task of e.g. a connection pool resp. obligation of the caller. If we would pass the connection via the constructor one would have to instantiate a new `SqlQueryExecutor` instance whenever a query is issued; afterwards the instance is right away going to be garbage collected: Seems to me somewhat superfluous. Could also remove that class if those utility methods aren't needed?

Reply:
Yes, I was thinking about more on having a `SQLClient` object that could take care of executing queries and statement hiding details about connection retrieval and stuff, but it can be build eventually afterwards on top of the executor. 
So this is ok for me :+1: